packages/ui/components/ui/sheet.tsx

```diff
@@ -58,23 +59,59 @@ interface SheetContentProps
 const SheetContent = React.forwardRef<
   React.ElementRef<typeof SheetPrimitive.Content>,
   SheetContentProps
->(({ side = 'right', className, children, closeSheet, ...props }, ref) => (
+>(({ className, children, closeSheet, ...props }, ref) => (
   <SheetPortal>
-    <SheetOverlay />
-    <SheetPrimitive.Content ref={ref} className={cn(sheetVariants({ side }), className)} {...props}>
-      {children}
-      <SheetPrimitive.Close
-        className="ring-offset-background focus:ring-ring data-[state=open]:bg-secondary absolute right-4 top-4 rounded-sm opacity-70 transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:pointer-events-none"
+    <SheetPrimitive.Content
+      ref={ref}
+      className={cn(
+        'data-[state=open]:animate-in data-[state=closed]:animate-out fixed z-50 gap-4 p-2 shadow-lg transition ease-in-out data-[state=closed]:duration-300 data-[state=open]:duration-500',
+        'md:data-[state=closed]:slide-out-to-right md:data-[state=open]:slide-in-from-right md:sm:max-w-md md:inset-y-0 md:right-0 md:h-full md:w-3/4 md:border-transparent',
+        'max-md:data-[state=closed]:slide-out-to-bottom max-md:data-[state=open]:slide-in-from-bottom bottom-0 h-max w-full'
+      )}
+    >
+      <div
+        className={cn(
+          'fixed h-full w-24 -translate-x-16 cursor-pointer rounded-lg px-4 py-6 opacity-75 transition-all duration-200 hover:-translate-x-14 hover:bg-[rgba(255,255,255,0.1)]'
+        )}
         onClick={closeSheet}
       >
-        <X className="h-4 w-4" />
-        <span className="sr-only">Close</span>
-      </SheetPrimitive.Close>
+        <SheetPrimitive.Close
+          className="rounded-lg focus:outline-none focus:ring-2 focus:ring-offset-2"
+          onClick={closeSheet}
+        >
+          <ChevronsDownIcon className="stroke-web3wallet-text h-8 w-8 -rotate-90 transform" />
+          <span className="sr-only">Close</span>
```

Comment:
Localise?

Reply:
Not really possible atm, I'd like to see about doing it when we add localization